src/couch_replicator/src/couch_replicator_doc_processor.erl

```diff
@@ -54,14 +54,21 @@
 -define(TS_DAY_SEC, 86400).
 -define(INITIAL_BACKOFF_EXPONENT, 64).
 -define(MIN_FILTER_DELAY_SEC, 60).
+-define(DEFAULT_START_DELAY_MSEC, 5000).
+-define(MIN_START_DELAY_MSEC, 500).
 
 -type filter_type() :: nil | view | user | docids | mango.
--type repstate() :: initializing | error | scheduled.
+-type repstate() :: initializing | error | scheduled | not_owner.
+
+-record(st, {
+    tref :: reference(),
+    scanner_pid :: pid()
+}).
 
 -record(rdoc, {
     id :: db_doc_id() | '_' | {any(), '_'},
     state :: repstate() | '_',
-    rep :: #rep{} | nil | '_',
```

Comment:
Since we're keeping the rep state around in `#rdoc{}` we won't be setting it to `nil`. Previously we used an optimization when we started a job and the scheduler took care of maintaining and updating the rep record, we set the doc processor `rep` filed to `nil`.